components/widgets/menu/menu_items/menu_item.tsx

```diff
@@ -12,6 +12,7 @@ export default function menuItem(Component: React.ComponentType<any>) {
         id?: string;
         icon?: React.ReactNode;
         text?: React.ReactNode;
+        handleItemClick?: (event: React.MouseEvent<HTMLElement>) => void;
```

Comment:
Why not `onClick`?

Reply:
I guess this is because it is colliding with the underneath components. But I think all the underneath components should implement the onClick. That would be more consistent from the interface perspective. Than having an extra `handleItemClick` and in some cases also an `onClick`.

Reply:
Yeah, you are right. It did have conflicts with the onClick prop of a component. 

Thinking about it now I might need to adjust this a bit, since I did not want it to have it being called twice when named onClick.

I will thinking a better solution. Thanks for pointing that out.

Reply:
@jespino I added a new solution. Could you re-review and maybe take into consideration (since you are more experienced with this component) if the changes might have unwanted side-effects? It could possibly be, but pretty unlikely, that someone planned for an event to bubble form the component. I just want to make sure that this solution is not breaking anything!